[Form] moved data trimming logic of TrimListener into StringUtil

issei-m opened the pull request:
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

In the small job which needs not to build a form (e.g. just process a value gotten from Request), I want to reuse the trim logic.

issei-m committed:
[Form] moved data trimming logic of TrimListener into StringUtil

dosten commented:
Why put into `StringUtil` and not in `FormUtil`? Also, the `@author` of the `StringUtil` class must be you.

issei-m commented:
@dosten I also thought so but I decided to put into a new class because FormUtil is singular form. Should this code be moved into FormUtil?

> the `@author` of the StringUtil class must be you.

`TrimListener` was introduced by @webmozart at first. So it would be better to be his name.

Tobion commented on src/Symfony/Component/Form/Util/StringUtil.php:
```diff
@@ -0,0 +1,45 @@
+    /**
+     * Returns the trimmed data.
+     *
+     * @param mixed $data
```
it should only accept a string

stof replied:
well, this would force to keep the is_string check in TrimListener though

Tobion replied:
which is way better in terms of single responsibility.

Tobion replied:
why would this method handle non-strings here. doesn't make sense. it's not even documented that non-strings are returned as-is.

Tobion replied:
also this will allow us to add scalar type hints later in php 7.

issei-m commented:
I've just updated PR. Now, StringUtil does not concern argument type.
BTW, it would be better to add my name to `@author` in a patch like this? (I just moved codes, though)

issei-m commented:
ping @stof @Tobion

Tobion commented on src/Symfony/Component/Form/Tests/Util/StringUtilTest.php:
```diff
@@ -0,0 +1,75 @@
+use Symfony\Component\Form\Util\StringUtil;
+
+/**
+ * @author Bernhard Schussek <bschussek@gmail.com>
```
test classes dont have this annotation

issei-m commented:
@Tobion fixed it.
And how about this?

> BTW, it would be better to add my name to @author in a patch like this? (I just moved codes, though)

xabbuh commented:
@issei-m As you just moved code around, keeping @webmozart as the author is the best.

issei-m commented:
@xabbuh Thanks.

Well then, all issues were resolved.
Ready to be merged?

webmozart commented:
About the authorship, you should include yourself as author as well. I'd be pretty confused if I'd find myself as only author of a class that I had never seen before ;)

I think this is fine, but it can't be merged into 2.7 as that branch is feature frozen.

issei-m commented:
@webmozart Thanks for your advice. I'll try to open a new PR soon.
